libcudacxx/include/cuda/std/__atomic/functions/cuda_ptx_derived.h

```diff
@@ -33,6 +31,183 @@ _LIBCUDACXX_BEGIN_NAMESPACE_STD
 
 #if defined(_CCCL_CUDA_COMPILER)
 
+template <class _Operand>
+using __cuda_atomic_enable_non_native_arithmetic =
+  typename enable_if<_Operand::__size <= 16 || _Operand::__op == __atomic_cuda_operand::_f, bool>::type;
```

Comment:
all `typename enable_if<>::type` should be `__enable_if_t<>`

It is crazy, but it is actually better for compile times